awscommons/v2/autoscaling.go

```diff
@@ -0,0 +1,221 @@
+package awscommons
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/autoscaling"
+	autoscalingTypes "github.com/aws/aws-sdk-go-v2/service/autoscaling/types"
+	"github.com/gruntwork-io/go-commons/collections"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/go-commons/logging"
+	"github.com/gruntwork-io/go-commons/retry"
+)
+
+// GetAsgByName finds the Auto Scaling Group matching the given name. Returns an error if it cannot find a match.
+func GetAsgByName(opts *Options, asgName string) (*autoscalingTypes.AutoScalingGroup, error) {
+	client, err := NewAutoScalingClient(opts)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	input := &autoscaling.DescribeAutoScalingGroupsInput{AutoScalingGroupNames: []string{asgName}}
+	output, err := client.DescribeAutoScalingGroups(opts.Context, input)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+	groups := output.AutoScalingGroups
+	if len(groups) == 0 {
+		return nil, errors.WithStackTrace(NewLookupError("ASG", asgName, "detailed data"))
+	}
+	return &groups[0], nil
+}
+
+// ScaleUp sets the desired capacity, waits until all the instances are available, and returns the new instance IDs.
+func ScaleUp(
+	opts *Options,
+	asgName string,
+	originalInstanceIds []string,
+	desiredCapacity int32,
+	maxRetries int,
+	sleepBetweenRetries time.Duration,
+) ([]string, error) {
+	logger := logging.GetProjectLogger()
```

Comment:
In a library func, the logger needs to be injected (see above for rationale). I recommend accepting the logger as part of the `Options` struct.